src/lib.rs

```diff
@@ -1502,7 +1502,12 @@ impl Connection {
 
         if b.cap() < payload_len {
```

Comment:
better to be `!=`? since payload_len may use the value in the packet.

Reply:
e.g. in the additional test below, if I set `payload_len = 20` (not 4096), `pipe.server.recv()` will return `CryptoFail`. Which is fine but if we change here to `!=` it will be caught here so we don't even need to try decrypt.

Reply:
`b.cap()` represents the amount of bytes remaining in the whole UDP datagram, which could potentially include additional coalesced packets or additional junk at the end (e.g. see padding mechanism in FIrefox).

Though, thinking about this more, we don't actually need an explicit check for this, as decryption will fail later anyway (with `BufferTooShort` which seems an appropriate error), so I'll just remove this check completely.

Reply:
See https://github.com/cloudflare/quiche/pull/460.